File: frontend/src/lib/api-response.ts
export interface ApiResponse<TData = unknown> {
  success: boolean;
  data?: TData;
  message?: string;
}

export interface ApiErrorResponse {
  success: false;
  message: string;
  errors?: Record<string, string[]>;
}

export interface ApiListResponse<TData> extends ApiResponse<TData[]> {
  data: TData[];
}

File: backend/Church.Domain/Common/ApiResponse.cs
namespace Church.Domain.Common;

public sealed class ApiResponse<T>
{
    public bool Success { get; set; }
    public string Message { get; set; } = string.Empty;
    public T? Data { get; set; }
    public string? Error { get; set; }

    public static ApiResponse<T> Ok(T? data, string message = "success") => new()
    {
        Success = true,
        Message = message,
        Data = data,
        Error = null
    };

    public static ApiResponse<T> Fail(string message, string? error = null) => new()
    {
        Success = false,
        Message = message,
        Data = default,
        Error = error ?? message
    };
}